__tests__/find-ruby.test.ts

```diff
@@ -1,74 +1,67 @@
+      // process.stderr.write('write2:' + line + '\n');
+    });
+  });
 
-describe('find-ruby', () => {
-  beforeAll(async () => {
-    await io.rmRF(toolDir);
-    await io.rmRF(tempDir);
+  afterEach(() => {
+    tcSpy.mockClear();
+    cnSpy.mockClear();
+    jest.clearAllMocks();
   });
 
-  afterAll(async () => {
-    try {
-      await io.rmRF(toolDir);
-      await io.rmRF(tempDir);
-    } catch {
-      console.log('Failed to remove test directories');
-    }
-  }, 100000);
+  //afterAll(async () => {}, 100000);
 
-  it('Uses version of ruby installed in cache', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '17.0.0', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    // This will throw if it doesn't find it in the cache (because no such version exists)
-    await findRubyVersion('17.0.0');
-  });
+  it('finds a version of ruby already in the cache', async () => {
+    tcSpy.mockImplementation(() => '/cache/ruby/2.7.0');
 
-  it('Uses 9.0.x version of ruby installed in cache ', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '9.0.1', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    // This will throw if it doesn't find it in the cache (because no such version exists)
-    await findRubyVersion('9.0.x');
+    let cacheDir: string = await cache.find('2.7.0');
+    expect(cacheDir).toBe(normalize('/cache/ruby/2.7.0/bin'));
   });
 
-  it('Uses 9.0 version of ruby installed in cache ', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '9.0.1', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    // This will throw if it doesn't find it in the cache (because no such version exists)
-    await findRubyVersion('9.0');
-  });
+  it('does not find a version of ruby not in the cache', async () => {
+    tcSpy.mockImplementation(() => '');
 
-  it('findRubyVersion throws if cannot find any version of ruby', async () => {
-    let thrown = false;
-    try {
-      await findRubyVersion('9.9.9');
-    } catch {
-      thrown = true;
-    }
-    expect(thrown).toBe(true);
+    let cacheDir: string = await cache.find('2.9.9');
+    expect(cacheDir).toBeFalsy();
   });
 
-  it('findRubyVersion adds ruby bin to PATH', async () => {
-    const rubyDir: string = path.join(toolDir, 'Ruby', '2.4.6', os.arch());
-    await io.mkdirP(rubyDir);
-    fs.writeFileSync(`${rubyDir}.complete`, 'hello');
-    await findRubyVersion('2.4.6');
-    const binDir = path.join(rubyDir, 'bin');
-    console.log(`binDir: ${binDir}`);
-    console.log(`PATH: ${process.env['PATH']}`);
-    expect(process.env['PATH']!.startsWith(`${binDir}`)).toBe(true);
+  it('finds a version in the cache and adds it to the path', async () => {
+    inSpy.mockImplementation(() => '2.7.0');
+    tcSpy.mockImplementation(() => '/cache/ruby/2.7.0');
+    await run();
```

Comment:
to make this work, I had to factor out run into main.  It worked if I imported setup-ruby directly at top of the file but then run would get called at import time causing a superflous error output that version spec is required leading to an annotation in the test log